src/mlpack/tests/BERT_tokenizer_test.cpp

```diff
@@ -0,0 +1,128 @@
+
+    // Running tokenization for each string in the given input vector.
+    for(int i = 0;i < input_list.size();i++)
+    {
+    	tokens = tokenizer.tokenize(input_list[i]);
+    	for(int j = 0;j < tokens.size();j++)
+    	{
+    		// Converting each token we got to string format and storing as temp_str.
+    		convertFromUnicode(tokens[j], temp_str);
+    		temp.push_back(temp_str);
+    	}
+    	tokenized_output_pred.push_back(temp);
+    	temp = {};
+    }
+
+    // Cheking if the predicted list of tokens matches with the actual list of tokens.
+    REQUIRE(tokenized_output_pred.size() == tokenized_output_true.size());
+    for(i = 0;i < tokenized_output_pred.size();i++)
+    {
+    	REQUIRE(tokenized_output_pred[i].size() == tokenized_output_true[i].size());
+    	for(int j = 0;j < tokenized_output_pred[i];j++)
+    		REQUIRE(tokenized_output_pred[i][j].size() == tokenized_output_true[i][j].size());
+    }
+}
+
+/**
+ * Testing if the tokenization performs correctly for a string
+ * containing alphabets and spaces.
+ */
+TEST_CASE("BERTTokenizer-BasicTests", "[BERTTokenizerTest]")
+{
+	std::vector<std::string> input_list = {"this test case is for simple cases",
+		                      "checkinging for spliting",
+		                      "hello how are you",
+		                      "helloabcdef checkup"};
+	std::vector<std::vector<std::string>> tokenized_output_true = 
+	{
+		{"this", "test", "case", "is", "for", "simple", "cases"},
+		{"checking", "##ing", "for", "split", "##ing"},
+		{"hello", "how", "are", "you"},
+		{"hello", "##ab", "##cd", "##ef", "check", "##up"}
+	};
+
+	BERTTokenizerRunner(input_list, tokenized_output_true);
+}
+
+/**
+ * Testing if the tokenization performs correctly for a string
+ * containing alphabets, spaces and punctuations.
+ */
+TEST_CASE("BERTTokenizer-PunctTests", "[BERTTokenizerTest]")
+{
+    std::vector<std::string> input_list = {"check check make it , gtab",
+                              "UNwant 00E9d,running",
+                              "hello how are you?!!"};
+    std::vector<std::vector<std::string>> tokenized_output_true = 
+    {
+        {"check", "check", "make", "it", ",", "gt", "##ab"},
+        {"un", "##wan", "##t", "00", "##e", "##9", "##d", ",", "running"}
+    };
+
+    BERTTokenizerRunner(input_list, tokenized_output_true);
+}
+
+/**
+ * Testing if the tokenization performs correctly for a string
+ * containing alphabets, spaces and punctuations.
+ */
+TEST_CASE("BERTTokenizer-CapitalLetterTests", "[BERTTokenizerTest]")
+{
+    std::vector<std::string> input_list = {"[CLS] hello How ARE You?",
+                              "[CLS] This is a sentence. [SEP] Hoping it works [MASK] Fingers crossed [SEP]"};
+    std::vector<std::vector<std::string>> tokenized_output_true = 
+    {
+        {"[CLS]", "hello", "how", "are", "you", "?"},
+        {"[CLS]", "this", "is", "a", "sentence", ".", "[SEP]", "hoping", "it", "works", "[MASK]", "fingers", "crossed", "[SEP]"}
```

Comment:
Many of these lines are too long---please be sure to make them 80 characters or fewer.